src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java

```diff
@@ -2565,6 +2565,31 @@ public void sendWithCustomObjectTriggersOnHandlerWithMessagePack()  {
         assertEquals((short) 5, (short) person.getT());
     }
 
+    @Test
+    public void throwFromOnHandlerRunsAllHandlers() {
+        AtomicReference<String> value1 = new AtomicReference<>();
+        AtomicReference<String> value2 = new AtomicReference<>();
+
+        MockTransport mockTransport = new MockTransport();
+        HubConnection hubConnection = TestUtils.createHubConnection("http://example.com", mockTransport);
+
+        hubConnection.on("inc", (param1) -> {
+            value1.set(param1);
+            throw new RuntimeException("throw from on handler");
+        }, String.class);
+        hubConnection.on("inc", (param1) -> {
+            value2.set(param1);
+        }, String.class);
+
+        hubConnection.start().timeout(1, TimeUnit.SECONDS).blockingAwait();
+        mockTransport.receiveMessage("{\"type\":1,\"target\":\"inc\",\"arguments\":[\"Hello World\"]}" + RECORD_SEPARATOR);
+        hubConnection.send("inc", "Hello World");
```

Comment:
Why is this necessary? Isn't `mockTransport.receiveMessage(...` what triggers the on callbacks?